### DISCUSSION TITLE
Create View -> Map

### DISCUSSION THREAD

**Discussion by Simola4theWicked** [nocodb/nocodb#7588] [Features]
Dear NocoDB Team,

I am a user of NocoDB, greatly appreciating the flexibility and functionality it offers for managing and visualizing data in versatile ways. As I work extensively with geospatial data, I have a feature request that I believe would significantly enhance the user experience and functionality of NocoDB, especially for those dealing with geographic information.

**Feature Request:**

I propose the introduction of a new view option titled "Map View" under the "Create View" section. This view would allow users to visualize their geodata points directly on a map within the NocoDB interface. Here's a brief outline of how it could work:

- Map View Creation: Within the "Create View" dropdown, alongside the existing options (Grid, Gallery, Kanban, etc.), a new option for "Map" is available. Selecting this option creates a map view of the dataset, automatically detecting and plotting geodata points based on specified geospatial fields in the dataset.

- Data Point Visualization: All data points are plotted on the map, allowing for an intuitive and comprehensive spatial overview of the dataset. This would utilize latitude and longitude fields or recognized geospatial formats (e.g., GeoJSON, POINT).

- Interactive Elements: Users can click on any data point on the map to open a new window or pop-up. This window would display detailed information about the selected data point, showing the rest of the data fields associated with it, enhancing data interaction and exploration.

- Use Case Enhancement: This feature would be particularly beneficial for datasets involving locations, such as real estate listings, environmental monitoring points, delivery routes, etc., providing a straightforward way to visualize and interact with geospatial data.

**Value Proposition:**

Integrating a Map View option directly into NocoDB would streamline workflows for users handling geographic data, eliminating the need for external mapping tools and allowing for seamless data exploration and analysis within a single platform.

Thank you for considering this feature request. I believe that the addition of a Map View would make NocoDB an even more powerful tool for a wide range of applications involving geospatial data.

**Comment by caffeinepickles** [comment]
Upvote! This would open up a whole lot of new appliances out of the box.

**Comment by o1lab** [comment]
hey @Simola4theWicked : sorry for replying late. Saw your note almost everywhere :)

We do 've some support for map view which was developed by one of community member. We will try to consider this in Q1 roadmap else we should have it in first half of Q2.

**Reply by Simola4theWicked** [reply]
Hey @o1lab 

thank you for replying.
Yeah i wasn't sure where to post this so i posted it everywhere.

Awesome to hear that it's going to be implemented at one point this year.
Could you possibly point me to the community developed solution? Can't find it anywhere and would love to have a look at it.

**Reply by cloopadoop** [reply]
Any update on this? :)

**Comment by Simola4theWicked** [comment]
Hello,

I wanted to follow up on a Feature Request I submitted earlier regarding the ability to display geolocations on a map within NocoDB.

Current Request:

- Displaying Geolocation entries on a map.

Additional Ideas:
After using NocoDB extensively, I believe that this feature could be expanded to include the following enhancements:

- Custom Icons: It would be incredibly useful if we could upload and assign custom icons to different geolocation entries. This would allow for better visualization and differentiation of various types of locations on the map.

- Route Mapping: Another powerful feature would be the ability to plot routes between multiple geolocations directly on the map. This would be beneficial for users needing to visualize and manage paths, deliveries, or travel routes.

I believe these additional functionalities would significantly enhance the utility of the geolocation feature and make NocoDB even more versatile for users working with spatial data.

Could you please let me know the current status of this feature request and whether the new ideas could be considered in future updates?

Thank you for your time and consideration!

**Comment by khydrogenous** [comment]
Any updates on Map view ? We just migrated our entire DB stack to Noco and need to know if the Geo Data field is worth the extra time and effort to migrate compared to Single line text: "City, State, Country". Guys you are doing God's work, this is a fantastic product.

**Comment by guillermoscript** [comment]
any updates?
